packages/datadog-plugin-aws-sdk/src/base.js

```diff
@@ -34,6 +59,12 @@ class BaseAwsSdkPlugin extends ClientPlugin {
   constructor (...args) {
     super(...args)
 
+    // subscribe to network channels for serverless environment peer.service propagation
+    if (this._tracerConfig?._isInServerlessEnvironment()) {
+      networkChannelsStart.forEach(ch => this.addSub(ch, () => {}))
+      networkChannelsEnd.forEach(ch => this.addSub(ch, setPeerServiceTag))
```

Comment:
Why are these in `aws-sdk` instead of in the corresponding plugins?

Reply:
The corresponding plugins do not have the necessary information at their span level to determine hostname (via things like awsRegion and awsService)